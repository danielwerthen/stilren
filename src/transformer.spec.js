import createTransformer, { parseKey } from './transformer';

describe('parseKey', () => {
  it('should parse properly', () => {
    const breakpoints = {
      alpha: 1,
      beta: 2,
      zeta: 3,
    };
    const result = parseKey(
      'alphaBetaZetaDaniel',
      Object.keys(breakpoints),
      breakpoints
    );
    expect(result).toEqual(['daniel', 3, 2, 1]);
  });
  it('should only parse on camelcase', () => {
    const breakpoints = {
      alpha: 1,
      beta: 2,
      zeta: 3,
    };
    const result = parseKey(
      'alphametaDaniel',
      Object.keys(breakpoints),
      breakpoints
    );
    expect(result).toEqual(['alphametaDaniel']);
  });
});

describe('transformer', () => {
  const breakpoints = {
    alpha: '(min-width: 100px)',
    beta: '(max-width: 800px)',
  };
  const pseudos = ['hover', 'focus'];
  const transformer = createTransformer({
    breakpoints,
    pseudos,
  });
  describe('without extensions', () => {
    const transform = transformer({});
    expect(
      transform({
        fontSize: '15px',
        alphaBetaFontSize: '16px',
        betaAlphaFontSize: '17px',
      })
    ).toMatchSnapshot();
  });
  describe('with extensions', () => {
    const transform = transformer({
      myAttribute() {
        return 5;
      },
      myFunction(value) {
        return {
          color: value,
          background: value,
        };
      },
    });
    expect(
      transform({
        fontSize: '15px',
        myAttribute: true,
        myFunction: 'red',
        alphaMyFunction: 'yellow',
        alphaAlphaMyFunction: 'red',
        alphaHoverMyAttribute: true,
      })
    ).toMatchSnapshot();
  });
});
