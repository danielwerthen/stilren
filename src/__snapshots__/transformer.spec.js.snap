// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "@media (max-width: 800px) and (min-width: 100px)": Object {
    "fontSize": "17px",
  },
  "fontSize": "15px",
}
`;

exports[` 2`] = `
Object {
  ":hover": Object {
    "@media (min-width: 100px)": Object {
      "myAttribute": 5,
    },
  },
  "@media (min-width: 100px)": Object {
    "background": "red",
    "color": "red",
  },
  "background": "red",
  "color": "red",
  "fontSize": "15px",
  "myAttribute": 5,
}
`;
